app/src/main/java/com/chouten/app/data/ModuleModel.kt

```diff
@@ -194,8 +173,25 @@ data class InfoResult(
         val description: String?,
         val image: String?,
     ) {
-        override fun toString() =
-            "{\"url\": \"$url\", \"number\": $number, \"title\": \"$title\", \"description\": \"$description\", \"image\": \"$image\"}"
+        override fun toString(): String{
+            val map = mutableMapOf<String, Any?>()
+            map["url"] = url;
+            map["number"] = number;
+
+            if(title != null){
+                map["title"] = title;
+            }
+
+            if(description != null){
+                map["description"] = description;
+            }
+
+            if(image != null){
+                map["image"] = image;
+            }
+
+            return JSONObject(map).toString();
```

Comment:
`Mapper.json.encodeToString`

Reply:
Couldn't figure this out either. Added a TODO.

Reply:
It's quite easy, we have already defined a json formatter in MorePage.kt:
```kt
json.encodeToString(serializer(), "<data class you want to encode to json here>")
```

Reply:
actually, now that I look at it, like Toby said we already have a Mapper class (`Mapper.json.encodeToString`) in Network.kt, so import that and then do the following:
```kt
Mapper.json.encodeToString(serializer(), "<data class you want to encode to json here>")
```